### ISSUE TITLE
Just getting this screen on bloodborne

### ISSUE THREAD

**Issue by xt234** [shadps4-emu/shadPS4#1055]
![Screenshot_20240924_101929](https://github.com/user-attachments/assets/fde6a4f5-444f-49c3-93f4-e1433721e69d)
anyone know how to fix this?

**Comment by Ein420** [comment]
similar issue here I keep getting a black screen

Arch Linux
Kernel 6.11.0-arch1-1
Ryzen 5 4600G
RX 6600
16 GB RAM
[shad_log.txt](https://github.com/user-attachments/files/17116352/shad_log.txt)

`[Kernel.Vmm] <Info> memory_management.cpp:sceKernelMapNamedFlexibleMemory:207: addr = 0x880000000, len = 0x4000, prot = 0x3, flags = 0x0
[Core.Linker] <Info> module.cpp:Start:72: Module started : libc
[Kernel.Pthread] <Info> thread_management.cpp:scePthreadMutexInit:450: name=SceLibc_Atexit, result=0
[Kernel.Pthread] <Info> thread_management.cpp:scePthreadMutexInit:450: name=SceLibc_Locale, result=0
[Kernel.Pthread] <Info> thread_management.cpp:scePthreadMutexInit:450: name=SceLibc_Malloc, result=0
[Kernel.Pthread] <Info> thread_management.cpp:scePthreadMutexInit:450: name=SceLibc_Stream, result=0
[Kernel.Pthread] <Info> thread_management.cpp:scePthreadMutexInit:450: name=SceLibc_Debug, result=0
[Kernel.Pthread] <Info> thread_management.cpp:scePthreadMutexInit:450: name=SceLibc_FileStdin, result=0
[Kernel.Pthread] <Info> thread_management.cpp:scePthreadMutexInit:450: name=SceLibc_FileStdout, result=0
[Kernel.Pthread] <Info> thread_management.cpp:scePthreadMutexInit:450: name=SceLibc_FileStderr, result=0
[Core.Linker] <Info> module.cpp:Start:72: Module started : libSceFace
[Core.Linker] <Info> module.cpp:Start:72: Module started : libSceSmart
[Core] <Error> stubs.cpp:CommonStub:41: Stub: _ZNSt8ios_base4InitC1Ev (nid: sqWytnhYdEg) called, returning zero to 0x93003b3c3
[Core] <Error> stubs.cpp:CommonStub:41: Stub: __cxa_atexit (nid: tsvEmnenz48) called, returning zero to 0x93003b3dc
[Core] <Error> stubs.cpp:CommonStub:41: Stub: _ZNSt6_WinitC1Ev (nid: -Bl9-SZ2noc) called, returning zero to 0x93003b3eb
[Core] <Error> stubs.cpp:CommonStub:41: Stub: __cxa_atexit (nid: tsvEmnenz48) called, returning zero to 0x92fffc074
[Core] <Error> stubs.cpp:CommonStub:41: Stub: _ZNSt8ios_base4InitC1Ev (nid: sqWytnhYdEg) called, returning zero to 0x930036b83
[Core] <Error> stubs.cpp:CommonStub:41: Stub: __cxa_atexit (nid: tsvEmnenz48) called, returning zero to 0x930036b9c
[Core] <Error> stubs.cpp:CommonStub:41: Stub: _ZNSt6_WinitC1Ev (nid: -Bl9-SZ2noc) called, returning zero to 0x930036bab
[Core] <Error> stubs.cpp:CommonStub:41: Stub: __cxa_atexit (nid: tsvEmnenz48) called, returning zero to 0x92fffc074
[Core] <Error> stubs.cpp:CommonStub:41: Stub: _ZNSt8ios_base4InitC1Ev (nid: sqWytnhYdEg) called, returning zero to 0x9300237d3
[Core] <Error> stubs.cpp:CommonStub:41: Stub: __cxa_atexit (nid: tsvEmnenz48) called, returning zero to 0x9300237ec
[Core] <Error> stubs.cpp:CommonStub:41: Stub: _ZNSt6_WinitC1Ev (nid: -Bl9-SZ2noc) called, returning zero to 0x9300237fb
[Core] <Error> stubs.cpp:CommonStub:41: Stub: __cxa_atexit (nid: tsvEmnenz48) called, returning zero to 0x92fffc074
[Core] <Error> stubs.cpp:CommonStub:41: Stub: _ZNSt8ios_base4InitC1Ev (nid: sqWytnhYdEg) called, returning zero to 0x92fffed93
[Core] <Error> stubs.cpp:CommonStub:41: Stub: __cxa_atexit (nid: tsvEmnenz48) called, returning zero to 0x92fffedac
[Core] <Error> stubs.cpp:CommonStub:41: Stub: _ZNSt6_WinitC1Ev (nid: -Bl9-SZ2noc) called, returning zero to 0x92fffedbb
[Core] <Error> stubs.cpp:CommonStub:41: Stub: __cxa_atexit (nid: tsvEmnenz48) called, returning zero to 0x92fffc074
[Core] <Error> stubs.cpp:CommonStub:41: Stub: _ZNSt8ios_base4InitC1Ev (nid: sqWytnhYdEg) called, returning zero to 0x92fffd223
[Core] <Error> stubs.cpp:CommonStub:41: Stub: __cxa_atexit (nid: tsvEmnenz48) called, returning zero to 0x92fffd23c
[Core] <Error> stubs.cpp:CommonStub:41: Stub: _ZNSt6_WinitC1Ev (nid: -Bl9-SZ2noc) called, returning zero to 0x92fffd24b
[Core] <Error> stubs.cpp:CommonStub:41: Stub: __cxa_atexit (nid: tsvEmnenz48) called, returning zero to 0x92fffc074
[Core] <Error> stubs.cpp:CommonStub:41: Stub: _ZNSt8ios_base4InitC1Ev (nid: sqWytnhYdEg) called, returning zero to 0x92fffc963
[Core] <Error> stubs.cpp:CommonStub:41: Stub: __cxa_atexit (nid: tsvEmnenz48) called, returning zero to 0x92fffc97c
[Core] <Error> stubs.cpp:CommonStub:41: Stub: _ZNSt6_WinitC1Ev (nid: -Bl9-SZ2noc) called, returning zero to 0x92fffc98b
[Core] <Error> stubs.cpp:CommonStub:41: Stub: __cxa_atexit (nid: tsvEmnenz48) called, returning zero to 0x92fffc074
[Core.Linker] <Info> module.cpp:Start:72: Module started : libSceHeadTracker
[Core] <Error> stubs.cpp:CommonStub:41: Stub: __cxa_atexit (nid: tsvEmnenz48) called, returning zero to 0x93fffc074
[Core] <Error> stubs.cpp:CommonStub:41: Stub: __cxa_atexit (nid: tsvEmnenz48) called, returning zero to 0x93fffc074
[Core.Linker] <Info> module.cpp:Start:72: Module started : libSceHand
[Core.Linker] <Info> module.cpp:Start:72: Module started : libSceFios2
[Core.Linker] <Info> module.cpp:Start:72: Module started : libSceS3DConversion
[Kernel.Vmm] <Warning> memory_management.cpp:sceKernelGetDirectMemorySize:19: called
[Kernel.Vmm] <Info> memory_management.cpp:sceKernelAllocateDirectMemory:48: searchStart = 0x0, searchEnd = 0x120000000, len = 0x7400000, alignment = 0x200000, memoryType = 0x0, physAddrOut = 0x0
[Kernel.Vmm] <Info> memory_management.cpp:sceKernelMapDirectMemory:177: called, redirected to sceKernelMapNamedDirectMemory
[Kernel.Vmm] <Info> memory_management.cpp:sceKernelMapNamedDirectMemory:147: addr = 0x0, len = 0x7400000, prot = 0x3, flags = 0x0, directMemoryStart = 0x0, alignment = 0x200000
[Lib.SysModule] <Error> sysmodule.cpp:sceSysmoduleLoadModule:47: (DUMMY) called module = ORBIS_SYSMODULE_APP_CONTENT
[Lib.AppContent] <Error> app_content.cpp:sceAppContentInitialize:246: (DUMMY) called
[Kernel.Pthread] <Error> thread_management.cpp:scePthreadAttrSetschedpolicy:284: policy=1 not supported by winpthreads
[Kernel.Pthread] <Error> thread_management.cpp:scePthreadAttrSetschedpolicy:284: policy=1 not supported by winpthreads
[Kernel.Pthread] <Info> thread_management.cpp:scePthreadCreate:1038: thread create name = Core.Logging.BackgroundStrategy
[Lib.UserService] <Warning> userservice.cpp:sceUserServiceInitialize:1124: (dummy) called
[Lib.SystemService] <Info> systemservice.cpp:sceSystemServiceParamGetInt:1894: called param_id 1
[Lib.NpManager] <Error> np_manager.cpp:sceNpSetContentRestriction:1974: (STUBBED) called
[Kernel.Vmm] <Warning> memory_management.cpp:sceKernelGetDirectMemorySize:19: called
[Kernel.Vmm] <Info> memory_management.cpp:sceKernelAllocateDirectMemory:48: searchStart = 0x0, searchEnd = 0x120000000, len = 0x20c00000, alignment = 0x4000, memoryType = 0x0, physAddrOut = 0x7400000
[Kernel.Vmm] <Info> memory_management.cpp:sceKernelMapDirectMemory:177: called, redirected to sceKernelMapNamedDirectMemory
[Kernel.Vmm] <Info> memory_management.cpp:sceKernelMapNamedDirectMemory:147: addr = 0x0, len = 0x20c00000, prot = 0x3, flags = 0x0, directMemoryStart = 0x7400000, alignment = 0x4000
[Kernel.Vmm] <Warning> memory_management.cpp:sceKernelGetDirectMemorySize:19: called
[Kernel.Vmm] <Info> memory_management.cpp:sceKernelAllocateDirectMemory:48: searchStart = 0x0, searchEnd = 0x120000000, len = 0x11400000, alignment = 0x4000, memoryType = 0x0, physAddrOut = 0x28000000
[Kernel.Vmm] <Info> memory_management.cpp:sceKernelMapDirectMemory:177: called, redirected to sceKernelMapNamedDirectMemory
[Kernel.Vmm] <Info> memory_management.cpp:sceKernelMapNamedDirectMemory:147: addr = 0x0, len = 0x11400000, prot = 0x3, flags = 0x0, directMemoryStart = 0x28000000, alignment = 0x4000
[Kernel.Vmm] <Warning> memory_management.cpp:sceKernelGetDirectMemorySize:19: called
[Kernel.Vmm] <Info> memory_management.cpp:sceKernelAllocateDirectMemory:48: searchStart = 0x0, searchEnd = 0x120000000, len = 0x9800000, alignment = 0x10000, memoryType = 0x0, physAddrOut = 0x39400000
[Kernel.Vmm] <Info> memory_management.cpp:sceKernelMapDirectMemory:177: called, redirected to sceKernelMapNamedDirectMemory
[Kernel.Vmm] <Info> memory_management.cpp:sceKernelMapNamedDirectMemory:147: addr = 0x0, len = 0x9800000, prot = 0x33, flags = 0x0, directMemoryStart = 0x39400000, alignment = 0x10000
[Kernel.Vmm] <Warning> memory_management.cpp:sceKernelGetDirectMemorySize:19: called
[Kernel.Vmm] <Info> memory_management.cpp:sceKernelAllocateDirectMemory:48: searchStart = 0x0, searchEnd = 0x120000000, len = 0x8ef00000, alignment = 0x10000, memoryType = 0x3, physAddrOut = 0x42c00000
[Kernel.Vmm] <Info> memory_management.cpp:sceKernelMapDirectMemory:177: called, redirected to sceKernelMapNamedDirectMemory
[Kernel.Vmm] <Info> memory_management.cpp:sceKernelMapNamedDirectMemory:147: addr = 0x0, len = 0x8ef00000, prot = 0x33, flags = 0x0, directMemoryStart = 0x42c00000, alignment = 0x10000`

**Comment by Vladimir-A** [comment]
Same
https://gist.github.com/Vladimir-A/6d7090138f26b837bbc066134e6db015

**Comment by GryfHub** [comment]
how do we know when this is fixed ?

**Comment by raphaelthegreat** [comment]
Linux cannot boot bloodborne without a patch on main

**Comment by junqueira200** [comment]
How can I make this patch?

**Comment by robyn-dressler** [comment]
@junqueira200

I made this [patch](https://gist.github.com/robyn-dressler/6b7570d12b57726ca72ae40428079770) to try and fix the black screen locally. I took the changes from the bb-hacks branch and updated them to work with 0.3.0 release. I was able to get past the black screen, but there are still issues with audio.

Just as a disclaimer, I barely understood any of the code I was modifying. I only made small changes to get @raphaelthegreat's workaround to compile with a more up to date version.
